### DISCUSSION TITLE
Graphics of the map on HD monitors must not be displayed 1x1 to the original format

### DISCUSSION THREAD

**Discussion by Dmitriy68** [ihhub/fheroes2#4832] [Q&A]
For normal use of the game, it is necessary that the graphics of the map on HD monitors are not displayed 1x1 to the original format, but somehow scaled.
My child and I couldn't use the game because all the map objects are too small.

**Comment by a1exsh** [comment]
Have you tried changing the game resolution to a bit lower than full HD? Something like 1024x768 should be very playable.

**Reply by Dmitriy68** [reply]
I was talking about the problem of not scaling the map. There is a change in the size of the overall frame of the game, but the game map does not scale inside it.
All map objects turn out to be too small.

**Reply by a1exsh** [reply]
Do you mean the minimap? What resolution are you actually using? Maybe make a screenshot? ;)

**Comment by Dmitriy68** [comment]
Here is an example - HD monitor 1920x1200

![Example_01](https://user-images.githubusercontent.com/96588876/147384707-38ef8410-a215-47bd-a5f7-6ea1f4cf70ff.jpg)

**Reply by a1exsh** [reply]
> Here is an example - HD monitor 1920x1200
> 
> ![Example_01](https://user-images.githubusercontent.com/96588876/147384707-38ef8410-a215-47bd-a5f7-6ea1f4cf70ff.jpg)

I'm still not sure if you're referring to the minimap in the top right corner, or to the actual adventure map.

In either case, setting the game resolution to a smaller one and turning into fullscreen mode (`F4` on the keyboard) should help — the whole game is scaled to the display resolution then.

**Comment by Dmitriy68** [comment]
No, I'm not discussing the minimap.
In general, I managed to understand the logic of primitive scaling of the game map. It can be adjusted by the size of the overall frame.
If you set the frame size to 1024x768, then the dimensions of map objects become minimally acceptable. But they are again smaller than in the original DOS version (as in DOSBox).
But any scale larger than 1024x768 makes objects microscopic. So, in general, this is a very big flaw in the interface.

**Reply by a1exsh** [reply]
> But any scale larger than 1024x768 makes objects microscopic. So, in general, this is a very big flaw in the interface.

I do not agree.  The whole point of supporting resolutions higher than the original game's 640x480 is to be able to fit more objects on the map.  If we scale the objects up, less of them will fit on one screen, defeating the purpose.

If objects "smaller than in the original" is problematic for you, you can always set the resolution to the same as in the original game — or at least don't set it higher than 1024x768 if that's uncomfortable for you. ;)

**Reply by oleg-derevenetz** [reply]
As for me, my Retina display supports in-game resolutions up to 2880x1800, and the game is perfectly playable in 1920x1200 resolution for example (however in 2880x1800 objects are really too small). I'd rather see more of the adventure map in the higher resolution (naturally, with objects getting smaller) instead of upscaling everything to always look as in original 640x480 (that's what DOSBOX do, because it simply has no other choice). This is nonsense to support higher resolutions if there is no benefit from it.
